### PR TITLE
fix: keep availability ranges on the same calendar day

### LINKED ISSUE #159: Availability scheduler allows end time before start
## Problem

A time range can end earlier in the clock than it starts. In the preview this shows up as **Monday 7:00 PM – 1:00 AM**: `1:00 AM` is treated as the same calendar day, so the range is invalid (and is not an overnight window).

## Repro

1. Open `/components/blocks/availability-scheduler`.
2. On an available day, set **start** to 7:00 PM and **end** to 1:00 AM.
3. The UI accepts the pair and `onChange` emits it.

## Expected

Same-day ranges only: end must be after start. The end picker should not list times at or before start; 7:00 PM → 1:00 AM should be rejected or clamped.

Image

### PR DESCRIPTION
**Pull request by invictusdhahri**
## Summary
- Fixes #159: the scheduler accepted ranges whose end is earlier than the start (e.g. 7:00 PM → 1:00 AM).
- Start/end pickers only list times that keep end after start; invalid pairs are clamped to the next valid slot.

## Test plan
- [ ] Open `/components/blocks/availability-scheduler`
- [ ] Set a day's start to 7:00 PM — 1:00 AM should not be selectable as the end
- [ ] Change end first to 1:00 AM, then start to 7:00 PM — range should clamp so end stays after start
- [ ] 9:00 AM – 5:00 PM still works
- [ ] `bun test tests/availability-scheduler.test.tsx`

### CODE REVIEW

File: components/motion/availability-scheduler/types.ts [hunk 1]
```diff
@@ -56,6 +56,32 @@ export function buildOptions(step: number): TimeOption[] {
   return out;
 }
 
+/** Same-day ranges only: end must be after start. 7:00 PM → 1:00 AM is rejected. */
+export function clampRange(
+  start: string,
+  end: string,
+  step: number,
+): { start: string; end: string } {
+  const last = 24 * 60 - step;
```
Comment (starc007) [inline components/motion/availability-scheduler/types.ts]: **[P2] Clamp against the actual generated options**

This assumes the final option is always `24h - step`, which only holds when `step` divides 1440. For example, with `step={50}`, the picker contains `22:30` and `23:20`, but clamping that already-valid range rewrites it to `22:20`–`23:10`; neither value exists in the picker. Please clamp against neighboring values from the generated options, or explicitly constrain and validate `step`.

File: components/motion/availability-scheduler/day-row.tsx [hunk 2]
```diff
@@ -167,7 +179,7 @@ export function DayRow({
                 <div className="min-w-0 flex-1 sm:max-w-[132px]">
                   <TimeSelect
                     value={r.start}
-                    options={options}
+                    options={startOptions(options, r.end)}
```
Comment (starc007) [inline components/motion/availability-scheduler/day-row.tsx]: **[P2] Keep previously emitted invalid values visible**

For a persisted range such as `19:00`–`01:00`, `startOptions` removes `19:00` and `endOptions` removes `01:00`. `SelectValue` only knows labels registered by rendered items, so both triggers display the generic “Select” placeholder and conceal the stored values. This fix targets values the previous component could emit, so please normalize incoming ranges or keep each current option rendered while requiring the next selection to be valid.

**Review by starc007 — CHANGES_REQUESTED** [review]
The same-day filtering is the right direction, but two edge cases can still leave the scheduler with invalid or concealed values:

- `clampRange` produces times outside the generated option set when `step` does not divide 1440.
- Previously persisted invalid ranges lose both visible labels because the selected values are filtered out.

Please also bump `blocks/availability-scheduler.updatedAt` in `lib/component-dates.ts` to the ship date, since this changes public behavior.

The scheduler tests, typecheck, lint, and registry validation pass. The current CI failure is an unrelated `AttachmentUpload` timeout; Vercel requires deployment authorization.

File: components/motion/availability-scheduler/types.ts (hunk 1 above)
Comment (invictusdhahri) [reply components/motion/availability-scheduler/types.ts]: Fixed in `e75ba97`. `clampRange` now snaps start/end onto the generated option grid and only walks to a neighboring option when the range is still invalid. `step={50}` keeps `22:30`–`23:20` as-is.

File: components/motion/availability-scheduler/day-row.tsx (hunk 2 above)
Comment (invictusdhahri) [reply components/motion/availability-scheduler/day-row.tsx]: Fixed in `e75ba97`. The current start/end stay in each picker so a persisted `19:00`–`01:00` still shows `7:00 PM` / `1:00 AM`. The next pick still has to be a same-day pair; `clampRange` runs on change.

**Comment by invictusdhahri** [comment]
Addressed the review on `e75ba97`:

- `clampRange` now snaps onto `buildOptions(step)` instead of assuming the last slot is `24h - step`.
- Persisted overnight values stay visible in both selects; the next selection still has to be same-day.
- `blocks/availability-scheduler.updatedAt` is `2026-08-23`.

File: components/motion/availability-scheduler/types.ts [hunk 3]
```diff
@@ -56,6 +56,82 @@ export function buildOptions(step: number): TimeOption[] {
...
+  end: string,
+  options: TimeOption[],
+): { start: string; end: string } {
+  const slots = options.map((o) => toMinutes(o.value));
+  if (slots.length === 0) return { start, end };
+  if (slots.length === 1) {
+    return { start: options[0].value, end: options[0].value };
+  }
+
+  let s = snapToOption(toMinutes(start), slots);
```
Comment (starc007) [inline components/motion/availability-scheduler/types.ts]: `updateRange` does not tell `clampRange` which endpoint changed, so both endpoints are snapped. With the existing `step={720}` case and default `09:00–17:00`, selecting `12:00` as the start calls this with `12:00–17:00`; the end snaps to `12:00`, then the fallback moves the selected start backward and emits `00:00–12:00`. The user’s selection is silently undone even though `12:00–17:00` is already a valid same-day range. Please preserve the endpoint the user selected and only move the opposite endpoint when the invariant requires it, then add an interaction test for this case.

**Review by starc007 — COMMENTED** [review]
One interaction edge case remains after the latest update.

File: components/motion/availability-scheduler/types.ts (hunk 3 above)
Comment (invictusdhahri) [reply components/motion/availability-scheduler/types.ts]: Fixed in `6fe3628`. `updateRange` now passes which endpoint changed. A valid same-day pair is left alone, so picking `12:00` as the start against a default `17:00` end on `step={720}` stays `12:00`–`17:00`. Only the opposite side moves when the range is invalid. Added the interaction test for that case.

**Comment by invictusdhahri** [comment]
Follow-up in `6fe3628`: `clampRange` now knows which endpoint the user just picked, so a valid `12:00`–`17:00` on the 12h step is no longer rewritten to `00:00`–`12:00`.

**Comment by invictusdhahri** [comment]
It should covered now!

**Comment by starc007** [comment]
Thank you for the thoughtful follow-ups and quick iterations, Amen. I added the final boundary recovery case in `976d195a` so legacy `23:30–00:00` ranges can be repaired directly from the pickers without deleting the row. The full check and all 354 tests pass locally. Appreciate the contribution!

**Review by starc007 — APPROVED** [review]
All requested scheduler range cases are addressed, including recovery from persisted boundary overnight values. Checks are green.
